File: dispatch/with_data/C#/dispatch_class.cs
using System;
using System.IO;
using System.Diagnostics;

// Interface for values
interface IValue {
    void Process();
    void Print();
}

// Implementation for int values
class IntValue : IValue {
    private int _value;

    public IntValue(int initial = 42) {
        _value = initial;
    }

    public void Process() {
        _value += 10; // Example: add 10 to int
    }

    public void Print() {
        Console.WriteLine($"Processed int: {_value}");
    }
}

// Implementation for float values
class FloatValue : IValue {
    private float _value;

    public FloatValue(float initial = 3.14f) {
        _value = initial;
    }

    public void Process() {
        _value *= 2.0f; // Example: multiply float by 2
    }

    public void Print() {
        Console.WriteLine($"Processed float: {_value}");
    }
}

// Implementation for double values
class DoubleValue : IValue {
    private double _value;

    public DoubleValue(double initial = 2.718281828459045) {
        _value = initial;
    }

    public void Process() {
        _value /= 2.0; // Example: divide double by 2
    }

    public void Print() {
        Console.WriteLine($"Processed double: {_value}");
    }
}

// Factory class to create appropriate value objects
static class ValueFactory {
    public static IValue CreateValue(string type) => type switch {
        "int" => new IntValue(),
        "float" => new FloatValue(),
        "double" => new DoubleValue(),
        _ => throw new ArgumentException($"Unsupported type: {type}")
    };
}

// Configuration reader class
static class Config {
    public static string ReadType(string filename) {
        if (!File.Exists(filename)) {
            throw new FileNotFoundException($"Failed to open config file: {filename}");
        }

        string type = File.ReadAllText(filename).Trim();
        return type;
    }
}

class Program {
    static void Main() {
        try {
            // Read configuration
            const string configFile = "config.txt";
            string type = Config.ReadType(configFile);

            // Create value object based on configuration
            IValue value = ValueFactory.CreateValue(type);

            // Start timing
            const int iterations = 1000000000;
            var stopwatch = Stopwatch.StartNew();

            // Process the value multiple times
            for (int i = 0; i < iterations; i++) {
                value.Process();
            }

            // Stop timing
            stopwatch.Stop();

            // Print results
            value.Print();
            Console.WriteLine($"Time taken for {iterations} iterations: {stopwatch.Elapsed.TotalSeconds:F6} seconds");

        } catch (Exception e) {
            Console.Error.WriteLine($"Error: {e.Message}");
            Environment.Exit(1);
        }
    }
}

File: dispatch/with_data/Go/dispatch_class.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
	"time"
)

// Value interface for polymorphic dispatch
type Value interface {
	Process()
	Print()
}

// IntValue implementation
type IntValue struct {
	value int32
}

func NewIntValue(initial int32) *IntValue {
	return &IntValue{value: initial}
}

func (v *IntValue) Process() {
	v.value += 10 // Example: add 10 to int
}

func (v *IntValue) Print() {
	fmt.Printf("Processed int: %d\n", v.value)
}

// FloatValue implementation
type FloatValue struct {
	value float32
}

func NewFloatValue(initial float32) *FloatValue {
	return &FloatValue{value: initial}
}

func (v *FloatValue) Process() {
	v.value *= 2.0 // Example: multiply float by 2
}

func (v *FloatValue) Print() {
	fmt.Printf("Processed float: %f\n", v.value)
}

// DoubleValue implementation
type DoubleValue struct {
	value float64
}

func NewDoubleValue(initial float64) *DoubleValue {
	return &DoubleValue{value: initial}
}

func (v *DoubleValue) Process() {
	v.value /= 2.0 // Example: divide double by 2
}

func (v *DoubleValue) Print() {
	fmt.Printf("Processed double: %f\n", v.value)
}

// ValueFactory creates value objects based on type string
func CreateValue(typeName string) (Value, error) {
	switch typeName {
	case "int":
		return NewIntValue(42), nil
	case "float":
		return NewFloatValue(3.14), nil
	case "double":
		return NewDoubleValue(2.718281828459045), nil
	default:
		return nil, fmt.Errorf("unsupported type: %s", typeName)
	}
}

// ReadType reads the type from config file
func ReadType(filename string) (string, error) {
	file, err := os.Open(filename)
	if err != nil {
		return "", fmt.Errorf("failed to open config file: %w", err)
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)
	if scanner.Scan() {
		return strings.TrimSpace(scanner.Text()), nil
	}

	if err := scanner.Err(); err != nil {
		return "", err
	}

	return "", fmt.Errorf("failed to read type from config file")
}

func main() {
	// Read configuration
	configFile := "config.txt"
	typeName, err := ReadType(configFile)
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error: %v\n", err)
		os.Exit(1)
	}

	// Create value object based on configuration
	value, err := CreateValue(typeName)
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error: %v\n", err)
		os.Exit(1)
	}

	// Start timing
	iterations := 1_000_000_000
	start := time.Now()

	// Process the value multiple times
	for i := 0; i < iterations; i++ {
		value.Process()
	}

	// Stop timing
	elapsed := time.Since(start)

	// Print results
	value.Print()
	fmt.Printf("Time taken for %d iterations: %.6f seconds\n", iterations, elapsed.Seconds())
}
